### PR TITLE
Add 'migrate user' subcommand.

### PR DESCRIPTION
**Pull request by atoaca**
For the password allocation on destination, the 'new_user_password' config option has been added.

### CODE REVIEW

File: coriolis_openstack_utils/conf.py [hunk 1]
```diff
@@ -117,6 +122,9 @@
 NEW_PHYSICAL_NETWORK_OPT = conf.DictOpt(
     "physical_network_mapping", default={},
     help="Mapping between physical networks on source and destination.")
+NEW_USER_PASSWORD_OPT = conf.StrOpt(
+    "new_user_password",
```
Comment (aznashwan) [inline coriolis_openstack_utils/conf.py]: Could you please rename this to `new_users_password` (plural on `users`)

File: coriolis_openstack_utils/cli/user.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                 "tenants on source to which the user originally had the "
+                 "admin role.")
+
+        return parser
+
+    def take_action(self, args):
+        # instantiate all clients:
+        source_client = conf.get_source_openstack_client()
+        destination_client = conf.get_destination_openstack_client()
+
```
Comment (aznashwan) [inline coriolis_openstack_utils/cli/user.py:61]: Please define `src_user_id = None` before the `try:`

File: coriolis_openstack_utils/resource_utils/users.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+def create_user(openstack_client, body):
+    body['password'] = CONF.destination.new_user_password
+    return openstack_client.keystone.users.create(**body).id
+
+
+def add_admin_roles(openstack_client, user_name, project_name_list):
+    for project_name in project_name_list:
+        LOG.info("Adding admin role to user '%s' in project '%s'"
+                 % (user_name, project_name))
+        openstack_client.add_admin_role_to_project(project_name, user_name)
```
Comment (aznashwan) [inline coriolis_openstack_utils/resource_utils/users.py]: Please either wrap this in a `try:-except keystone.NotFound:` and LOG if you can't find the tenant, or simply check for the tenant's existence before trying to add a role for the user in it.

File: coriolis_openstack_utils/resource_utils/users.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+def get_user_admin_tenants(openstack_client, user_id, admin_role_name='admin'):
+    admin_roles = [r for r in openstack_client.keystone.roles.list()
+                   if r.name == admin_role_name]
+
+    if not admin_roles:
+        raise Exception(
+            "Could not locate admin role named '%s' on destination" % (
+                admin_role_name))
+    elif len(admin_roles) > 1:
```
Comment (aznashwan) [inline coriolis_openstack_utils/resource_utils/users.py]: You can't have multiple roles named the same so this check is redundant.

File: migration-config.conf.sample [hunk 5]
```diff
@@ -38,6 +38,9 @@ new_tenant_name_format = "%(original)s-Migrated"
 new_secgroup_name_format = "%(original)s-Migrated"
 new_subnet_name_format = "%(original)s-Migrated"
 new_network_name_format = "%(original)s-Migrated"
+new_user_name_format = "%(original)s-Migrated"
```
Comment (aznashwan) [inline migration-config.conf.sample]: Could you please just leave the default as `%(original)s` (there's no real reason someone would want to have a different username, but having the option is nice, so the best route would be to just make the default case have the same name)

**Review by aznashwan — COMMENTED** [review]
LGTM just some minor nits.

**Comment by atoaca** [comment]
retested.

**Review by aznashwan — APPROVED** [review]
LGTM

**Pull request merged**